controller/service.go

```diff
@@ -148,7 +148,7 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 	}
 
 	pool := c.ips.Pool(key)
-	if pool == "" || c.pools[pool] == nil {
+	if c.isEmpty(pool) {
```

Comment:
I meant a method of pools, so we can call 
`if pool == "" || c.pools.IsEmpty()`